core/src/main/java/io/grpc/ServerInterceptors.java

```diff
@@ -80,6 +84,50 @@ public static ServerServiceDefinition intercept(ServerServiceDefinition serviceD
     return serviceDefBuilder.build();
   }
 
+  /**
+   * Create a new {@code ServerServiceDefinition} whose {@link MethodDescriptor} serializes to
+   * and from byte[] for all methods.  The {@code ServerCallHandler} created will automatically
+   * convert back to the original types for request and response before calling the existing
+   * {@code ServerCallHandler}.  Calling this method combined with the intercept methods will
+   * allow the developer to choose whether to intercept messages of byte[], or the modeled types
+   * of their application.
+   *
+   * @param serviceDef the service definition to convert messages to byte[]
+   * @return a wrapped version of {@code serviceDef} with the byte[] conversion applied.
+   */
+  public static ServerServiceDefinition useByteArrayMessages(
+      final ServerServiceDefinition serviceDef) {
+    final MethodDescriptor.Marshaller<byte[]> marshaller =
```

Comment:
That's reasonable, and maybe plays into the comment below about a more generic method that actually takes a `MethodDescriptor` or `Marshaller<T>` here.  The downside of passing the raw InputStream through is that once read, it's gone(unless I can mark it, though not guaranteed), so I have to replace it for anyone downstream that wants to use it.  This is a significant portion of what drove me in the direction I ended up picking here.  It also requires me to read the stream twice if I have 2 interceptors that want access to bytes.  If instead of InputStream it was a zero copy and multiple readable object like a ByteBuf that may work better since I can just slice a chunk of bytes out of the middle without any overhead and allow downstream readers to still be able to read it.